src/helpers/OracleMonitorConfig.js

```diff
@@ -0,0 +1,61 @@
+import { logger } from "./logger.js";
+import { readJSONFile } from "./utils.js";
+import { getAmountsIn, getOraclesInvitations } from "./chain.js";
+
+export class OracleMonitorConfig {
+  /**
+   * Constructor for the class
+   * @param {string} filePath file path of the oracle config
+   */
+  constructor(filePath) {
+    try {
+      this.oracles = readJSONFile(filePath);
+      this.amountsIn = {};
+      this.validate();
+      //get invitations
+      this.getInvsForOracles();
+    } catch (err) {
+      logger.error(
+        "Cannot load OracleMonitorConfig from " + filePath + ": " + err
+      );
+    }
+  }
+
+  /**
+   * Function to validate oracle configs
+   */
+  validate() {
+    for (let oracle in this.oracles) {
+      let currentOracle = this.oracles[oracle];
+      // If no oracle name
+      if (!("oracleName" in currentOracle)) {
+        throw new Error("No oracleName in oracle details");
+      }
+    }
+  }
+
+  /**
+   * Function to get feed invitations for multiple oracles
+   */
+  async getInvsForOracles() {
+    // For each oracle
+    for (let oracle in this.oracles) {
+      let invitations = await getOraclesInvitations(oracle);
+
+      // For each invitation
```

Comment:
for each invitation what?

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/29244ced3c6d34785c4643507480179d98d7d8c6